pkg/state/appender.go

```diff
@@ -764,6 +771,19 @@ func (a *txAppender) handleExchange(tx proto.Transaction, info *fallibleValidati
 			scriptsRuns++
 		}
 	}
+	// check attachment in order
+	txStateSnapshotActivated, err := a.stor.features.newestIsActivated(int16(settings.TransactionStateSnapshot))
+	if err != nil {
+		return nil, err
+	}
+	if !txStateSnapshotActivated {
+		if checkAttachment(exchange.GetOrder1()) {
+			return nil, errors.New("Attachment field for orders is not supported yet")
+		}
+		if checkAttachment(exchange.GetOrder2()) {
+			return nil, errors.New("Attachment field for orders is not supported yet")
+		}
+	}
```

Comment:
Actully, you can combine this conditions into one. For instance
`!txStateSnapshotActivated && (checkAttachment(exchange.GetOrder1()) || checkAttachment(exchange.GetOrder2()))`. But that's on your ch